pkg/controller/controller.go

```diff
@@ -104,11 +111,15 @@ func (c *PodController) doTheMagic(cur interface{}, keepSuccessHours int, keepFa
 	switch podObj.Status.Phase {
 	case v1.PodSucceeded:
 		if keepSuccessHours == 0 || (keepSuccessHours > 0 && executionTimeHours > float32(keepSuccessHours)) {
-			c.deleteObjects(podObj, createdMeta, dryRun)
+			c.deleteObjects(podObj, parentJobName, dryRun)
 		}
 	case v1.PodFailed:
 		if keepFailedHours == 0 || (keepFailedHours > 0 && executionTimeHours > float32(keepFailedHours)) {
-			c.deleteObjects(podObj, createdMeta, dryRun)
+			c.deleteObjects(podObj, parentJobName, dryRun)
+		}
+	case v1.PodPending:
+		if keepPendingHours == 0 || (keepPendingHours > 0 && executionTimeHours > float32(keepPendingHours)) {
```

Comment:
In  case of pending we should not even consider the case `keepPendingHours == 0`. It could result in chaos of deleting jobs during scheduling.